drivers/net/native_sim_errno.c

```diff
@@ -0,0 +1,123 @@
+/**
```

Comment:
Sidenote: This is so generic that I'm thinking we should make it common and provided by the native simulator code.
(it can start living here though)

Reply:
Left it here for now, but it definitely makes sense to move it to native_simulator code whenever there will be a need to reuse it in other context.